Purchases/Identity/RCPurchaserInfoManager.m

```diff
@@ -0,0 +1,196 @@
+//
+// Created by Andrés Boedo on 1/4/21.
+// Copyright (c) 2021 Purchases. All rights reserved.
+//
+
+#import "RCPurchaserInfoManager.h"
+#import "RCPurchaserInfo+Protected.h"
+#import "RCDeviceCache.h"
+#import "RCLogUtils.h"
+#import "RCBackend.h"
+#import "RCSystemInfo.h"
+@import PurchasesCoreSwift;
+
+NS_ASSUME_NONNULL_BEGIN
+
+
+@interface RCPurchaserInfoManager ()
+
+@property (nonatomic, nullable) RCPurchaserInfo *lastSentPurchaserInfo;
+@property (nonatomic) RCOperationDispatcher *operationDispatcher;
+@property (nonatomic) RCDeviceCache *deviceCache;
+@property (nonatomic) RCBackend *backend;
+@property (nonatomic) RCSystemInfo *systemInfo;
+
+@end
+
+@implementation RCPurchaserInfoManager
+
+@synthesize delegate = _delegate;
+
+- (void)setDelegate:(nullable id <RCPurchaserInfoManagerDelegate>)delegate {
+    _delegate = delegate;
+}
+
+- (instancetype)initWithOperationDispatcher:(RCOperationDispatcher *)operationDispatcher
+                                deviceCache:(RCDeviceCache *)deviceCache
+                                    backend:(RCBackend *)backend
+                                 systemInfo:(RCSystemInfo *)systemInfo {
+    self = [super init];
+    if (self) {
+        self.operationDispatcher = operationDispatcher;
+        self.deviceCache = deviceCache;
+        self.backend = backend;
+        self.systemInfo = systemInfo;
+    }
+
+    return self;
+}
+
+- (void)cachePurchaserInfo:(RCPurchaserInfo *)info forAppUserID:(NSString *)appUserID {
+    if (info) {
+        NSDictionary *infoAsJSONDict = info.JSONObject;
+        if ([NSJSONSerialization isValidJSONObject:infoAsJSONDict]) {
```

Comment:
this check is new. previously, if there was an error parsing the JSON (for example if the json has a value like `infinity`), the app would crash. 
the `jsonError` bit only helps for errors like incomplete stream and stuff like that, but it for some reason doesn't actually cover what would happen with an invalid json.